AP_Periph: Add BirdCANdy device

Maarrk opened the pull request:
Adds a tested configuration for a minimal open source AP_Periph target that can be soldered onto other PCBs or used standalone. Details of the hardware are in the [BirdCANdy repository](https://github.com/FlyfocusUAV/Bird-CANdy)

tridge commented on libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32F412Cx.py:
```diff
@@ -0,0 +1,347 @@
+#!/usr/bin/env python
```
you should be able to use the existing STM32F412Rx.py instead of adding a new one. We could rename it to STM32F412xx.py. The extra pins that the Cx don't have do no harm.

tridge commented on libraries/AP_HAL_ChibiOS/hwdef/scripts/STM32F412Cx.py:
```diff
@@ -0,0 +1,347 @@
+# MCU parameters
+mcu = {
+    # location of MCU serial number
+    'UDID_START' : 0x1FFF7A10,
```
this should come from the chibios headers now

Maarrk replied:
Fixed by removing the file

tridge reviewed (changes requested):
nice to see another node option! 
please change commit message to start with "HAL_ChibiOS:". If you don't know how to do that I can do it for you.
also let me know if there is some reason the existing STM32F412Rx.py doesn't work for the Cx

Maarrk commented:
I changed the commit message as requested, and removed `STM32F412Cx.py`.

I wasn't sure if you want me to rename `...Rx` to `...xx` and change it in all hwdefs that use it. If you confirm that, I can also do this in this PR.

When I did a git compare between `...Rx.py` and `...Cx,py` I found out that [the script I contributed some time ago to make these files](https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_HAL_ChibiOS/hwdef/scripts/af_parse_cubemx.py) sorts the pins in a slightly different order, that's why I didn't realize the Cx is just a subset. Sorry for the mistake, I probably should fix the script in another PR?

tridge commented:
@Maarrk I've rebased your PR and changed your board ID to 1044 as 1042 is already taken and 1043 is pending in another PR. I've added added your board to the autobuild list
Unfortunately I couldn't push over your branch as you didn't allow for maintainer updates when you created the PR
Can you push this over your PR branch?
  https://github.com/tridge/ardupilot/tree/pr-periph-birdcandy

tridge commented:
> I wasn't sure if you want me to rename `...Rx` to `...xx` and change it in all hwdefs that use it. If you confirm that, I can also do this in this PR.

yes, though it can be a separate PR so as not to hold up your board going in

Maarrk commented:
@tridge I believe it's correct now. 

Sorry for not letting you do edits, I must have missed the option this time. Will pay more attention in the future.

peterbarker commented:
@Maarrk sorry, this now needs a rebase.

I attempted to do this for you, but received:
```
pbarker@bluebottle:~/rc/ardupilot((HEAD detached from FlyfocusUAV/pr-periph-birdcandy))$ git push -f FlyfocusUAV HEAD:pr-periph-birdcandy 
ERROR: Permission to FlyfocusUAV/ardupilot.git denied to peterbarker.
fatal: Could not read from remote repository.
```

so it's on you now :-)

Maarrk commented:
@peterbarker I rebased, think it should merge now. It seems that AP_Periph is a very busy part of github at the moment 😅
